tools/shared.py

```diff
@@ -2356,6 +2356,11 @@ def get_subresource_location(path, data_uri=None):
     else:
       return os.path.basename(path)
 
+  # Strips all preprocessor directives from a string of JavaScript
+  @staticmethod
+  def strip_preprocessor_directives(s):
+    return re.sub('#if .*?\n.*?\n#endif', '', s, flags=re.DOTALL)
```

Comment:
i was thinking that we'd implement the operations too, and not just strip them all out. That is,
````
#if SINGLE_FILE
..
#endif
````
would fine the if, see that the var it checks is `SINGLE_FILE`, then check in `shared.Settings` if that is set, and remove or not remove the code accordingly.

What are the conditions we will need here? I assumed SINGLE_FILE would be one of them, is that wrong?